packages/oasgraph/src/index.ts

```diff
@@ -447,3 +449,75 @@ function sortOperations(op1: Operation, op2: Operation): number {
     }
   }
 }
+
+/**
+ * Ensures that the options are valid
+ */
+function preliminaryChecks(
+  options: InternalOptions,
+  data: PreprocessingData,
+  oass: Oas3[]
+): void {
+  // Check if OASs have unique titles
+  const titles = oass.map(oas => {
+    return oas.info.title
+  })
+
+  // Find duplicates among titles
+  new Set(
+    titles.filter((title, index) => {
+      return titles.indexOf(title) !== index
+    })
+  ).forEach(title => {
+    handleWarning({
+      typeKey: 'MULTIPLE_OAS_SAME_TITLE',
+      culprit: title,
+      data,
+      log: translationLog
+    })
+  })
```

Comment:
Could this not be much easier? I.e., use `const titleSet = new Set(oass.map(oas => oas.info.title)` and then `if (oass.length > titleSet.size) { // duplicate!`?

Reply:
Well, you would not have the `culprit` then...